spec/requests/users_controller_spec.rb

```diff
@@ -5273,6 +5273,84 @@ def users_found
     end
   end
 
+  describe "#lookup" do
+    fab!(:user_1) { Fabricate(:user) }
+    fab!(:user_2) { Fabricate(:user) }
+    fab!(:user_3) { Fabricate(:user) }
+
+    before { sign_in(user) }
+
+    it "lookups users by username" do
+      get "/u/lookup/users.json",
+          params: {
+            usernames: [user_1.username, user_2.username, user_3.username],
+          }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+      expect(users.length).to be(3)
+
+      ids = users.map { |user| user["id"] }
+      expect(ids).to include(user_1.id)
+      expect(ids).to include(user_2.id)
+      expect(ids).to include(user_3.id)
+    end
+
+    it "ignores non-existent usernames" do
+      get "/u/lookup/users.json",
+          params: {
+            usernames: [user_1.username, "unknown_username", "another_unknown_username"],
+          }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+      expect(users.length).to be(1)
+      expect(users[0]["id"]).to be(user_1.id)
+    end
+
+    it "returns users with ids, names and avatar_templates" do
+      get "/u/lookup/users.json", params: { usernames: [user_1.username] }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+
+      expect(users.length).to be(1)
+      expect(users[0]["id"]).to be(user_1.id)
+      expect(users[0]["username"]).to eq(user_1.username)
+      expect(users[0]["name"]).to eq(user_1.name)
+      expect(users[0]["avatar_template"]).to eq(user_1.avatar_template)
+    end
+
+    it "returns users with user status if status is enabled in settings" do
+      SiteSetting.enable_user_status = true
+      user_1.set_status!("off to dentist", "tooth")
+      user_1.reload
+
+      get "/u/lookup/users.json", params: { usernames: [user_1.username] }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+      expect(users.length).to be(1)
+      status = users[0]["status"]
+
+      expect(status["description"]).to eq(user_1.user_status.description)
+      expect(status["emoji"]).to eq(user_1.user_status.emoji)
+    end
+
+    it "returns users without user status if status is disabled in settings" do
+      SiteSetting.enable_user_status = false
+      user_1.set_status!("off to dentist", "tooth")
+      user_1.reload
```

Comment:
Is there any way we can avoid having to do this reload? This kind of knowledge is painful when we write specs.

Reply:
Oh, actually turned out we don't need to do it anymore, so I removed this `reload` call (see 7bee17f60dbfc26ee06f080d4f9f8bd2bfd76d67). We needed to do it in the past for some reason, but now it's fixed.